tests/parametric/test_span_links.py

```diff
@@ -0,0 +1,190 @@
+import pytest
+
+from ddapm_test_agent.trace import Span
+from ddapm_test_agent.trace import Trace
+from ddapm_test_agent.trace import root_span
+from utils.parametric.spec.trace import SAMPLING_PRIORITY_KEY, ORIGIN
+from utils.parametric.spec.trace import find_span
+from utils.parametric.spec.trace import find_trace_by_root
+from utils.parametric.spec.trace import span_has_no_parent
+from utils.parametric.headers import make_single_request_and_get_inject_headers
+from utils.parametric.test_agent import get_span
+from utils import bug, context, scenarios
+from utils.parametric._library_client import Link
+
+
+@scenarios.parametric
+class Test_Span_Links:
+    def test_span_started_with_link(self, test_agent, test_library):
+        """Test adding a span link created from another span.
+        This tests the functionality of "create a direct link between two spans
+        given two valid span (or SpanContext) objects" as specified in the RFC.
+        """
+        with test_library:
+            with test_library.start_span("root") as parent:
+                with test_library.start_span(
+                    "child",
+                    parent_id=parent.span_id,
+                    links=[Link(parent_id=parent.span_id, attributes={"foo": "bar", "array": ["a", "b", "c"]})],
```

Comment:
Not sure if you want to tackle this in this PR or a follow-up one, but I don't see any testing of attributes where the value is a Number or a Boolean

Reply:
Makes sense probably, in PHP everything is just stuffed into the same array and gets just stringified, hence I haven't thought much about that.

Will add :-)